GVRf/Framework/src/org/gearvrf/scene_objects/GVRVideoSceneObject.java

```diff
@@ -167,8 +189,16 @@ public MediaPlayer getMediaPlayer() {
      * @param mediaPlayer
      *            An Android {@link MediaPlayer}
      */
-    public void setMediaPlayer(MediaPlayer mediaPlayer) {
-        mVideo.setMediaPlayer(mediaPlayer);
+    public void setMediaPlayer(final MediaPlayer mediaPlayer) {
+        if (mVideo == null) {
```

Comment:
Having a setDataSource() in GVRVideoSceneObject.java would make things a lot easier. The call could internally create a media player and set its data source.  

This in turn eliminates the need for the createMediaPlayer api and also makes the scripting sample simpler. What do you think?